flash/nuke/nuke.c

```diff
@@ -27,12 +27,11 @@
 
 int main() {
     uint flash_size_bytes;
-#ifndef PICO_FLASH_SIZE_BYTES
-#warning PICO_FLASH_SIZE_BYTES not set, assuming 16M
-    flash_size_bytes = 16 * 1024 * 1024;
-#else
-    flash_size_bytes = PICO_FLASH_SIZE_BYTES;
-#endif
+    uint8_t txbuf[4];
+    uint8_t rxbuf[4];
+    txbuf[0] = 0x9f;
+    flash_do_cmd(txbuf, rxbuf, 4);
```

Comment:
Is this guaranteed to work with **every** variety of SPI Flash chip that might be used with RP2040 or RP2350?

Reply:
> I'm not an expert on this matter.

Me neither! The only reason I asked, was because I assumed that if it _was_ universally supported, then there'd already be a wrapper-function for it in pico-sdk? :thinking: 
But that's merely a hunch, I could be **entirely** wrong! :speak_no_evil:

Reply:
This suggests for an FPGA situation with NOR flash that two possible command sets exist, but I've no idea if that first option is supported by RP2s.
https://electronics.stackexchange.com/a/673366

Reply:
Okay, so digging deeper, yes the Serial Flash Discovery Protocol spec also mandates the backwards compatible support for 0x9f command, or at least as far as my limited research indicates.

Reply:
cc @Wren6991 who i believe has said in the past that such a command is not universally supported

Reply:
> That said it’s kinda up to the vendor to verify flash_nuke works on their board and - if not - supply a customised alternative.

:joy: :rofl: :joy: :rofl: :joy: :rofl: 

If every vendor provided even _half_ of the support that Pimoroni did, the world would be a much better place :grinning: